homeassistant/components/nuki/__init__.py

```diff
@@ -88,6 +127,31 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         sw_version=info["versions"]["firmwareVersion"],
     )
 
+    hass_url = get_url(hass)
+    # Check if hass_url is http
+    if hass_url.lower().startswith("http://"):
```

Comment:
This is not how this check should be done. `get_url` provides parameters to set your URL requirements and will throw when it cannot meet that.

See: <https://developers.home-assistant.io/docs/instance_url>

However, as per above by pvizeli, it should not handle/build this by itself.

Reply:
Added some additional parameters to the `get_url` function, but I don't see how we could remove the HTTP check here.

Reply:
I removed the issue registry stuff.